yarn.lock

```diff
@@ -0,0 +1,1728 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```

Comment:
Seems like this wouldnt need to be in Git. But I havent used yarn before.

Reply:
it does. this ensures that anyone else who clones the repo will have the same versions of every dependency. It's been awhile since I looked at what makes this different from package.json, but it's definitely needed

Reply:
huh. Insteresting. usually *.lock files are for locking the filesystem temporarily for a process to do stuff. Like with git this happens. (You notice it when you cancel git or it crashes and startsto complain. 

Interesting that they use that convention.